flexmeasures/data/models/parsing_utils.py

```diff
@@ -7,18 +9,26 @@
 
 
 def parse_source_arg(
-    source: Optional[
-        Union[DataSource, List[DataSource], int, List[int], str, List[str]]
-    ]
-) -> Optional[List[DataSource]]:
-    """Parse the "source" argument by looking up DataSources corresponding to any given ids or names."""
+    source: DataSource
+    | int
+    | str
+    | Sequence[DataSource]
+    | Sequence[int]
+    | Sequence[str]
+    | None,
+) -> list[DataSource] | None:
+    """Parse the "source" argument by looking up DataSources corresponding to any given ids or names.
+
+    Passes None as is (i.e. no source argument is given).
+    Accepts ids and names as list or tuples, always converting them to a list.
+    """
     if source is None:
         return source
-    if not isinstance(source, list):
+    if isinstance(source, (DataSource, str, int)):
         sources = [source]
     else:
         sources = source
-    parsed_sources: List[DataSource] = []
+    parsed_sources: list[DataSource] = []
```

Comment:
Is using `list` allowed since a particular Python version?

Reply:
3.9

Reply:
I know we are moving away from actively supporting it, but I also don't feel we should actively destroy 3.8 support at this point. Wouldn't this fail under 3.8?

Reply:
I was under the impression that the  `__future__` import takes care of compatibility. Is this not the case?

Reply:
I run 3.8. Let me run the tests in this branch ...

Reply:
They passed :)